sdks/python/apache_beam/runners/interactive/display/pcoll_visualization.py

```diff
@@ -236,6 +239,30 @@ def continuous_update_display():  # pylint: disable=unused-variable
   return None
 
 
+def visualize_computed_pcoll(
+    pcoll_name: str, pcoll: beam.pvalue.PValue) -> None:
+  """A simple visualize alternative.
+
+  When the pcoll_name and pcoll pair identifies a watched and computed
+  PCollection in the current interactive environment without ambiguity, an
+  ElementStream can be built directly from cache.
+  """
+  try:
+    pipeline = ie.current_env().user_pipeline(pcoll.pipeline)
+    cache_manager = ie.current_env().get_cache_manager(pipeline)
+    cache_key = CacheKey.from_pcoll(pcoll_name, pcoll).to_str()
+    reader, _ = cache_manager.read('full', cache_key)
+    visualize(
+        ElementStream(pcoll, pcoll_name, cache_key, float('inf'), float('inf')),
+        element_type=pcoll.element_type)
```

Comment:
Please move this logic to the RecordingManager class as a `read(pcoll: PValue)` method. It makes most sense to bundle as much logic about reading from cache and stream manipulation into that class so that this code isn't duplicated everywhere.

Reply:
Acked, will move it to RecordingManager.